react-accordion-component/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+import { AccordionItem } from './AccordionItem';
+
+export type Topic = {
+  title: string;
+  content: string;
+};
+
+type Props = {
+  topics: Topic[];
+};
+
+export function Accordion({ topics }: Props) {
+  const [currentIndex, setCurrentIndex] = useState<number | null>(null);
```

Comment:
If you use `undefined` as your "none is current" state, then you can just write:
`useState<number>();`

The default value for useState is `undefined`, and TypeScript will automatically add `| undefined` to the type if there are no arguments passed to `useState`. That is, with the above code, the type of `currentIndex` will be `number | undefined`.

Then, when you set the current index back to "none", you can just call `setCurrentIndex();` because once again, the default value is `undefined`.